README.md

```diff
@@ -188,6 +188,75 @@ pip install 'apache-airflow[postgres,google]==3.0.0' \
 For information on installing provider distributions, check
 [providers](http://airflow.apache.org/docs/apache-airflow-providers/index.html).
 
+## Setting up Apache Airflow on Apple Silicon (Mac M1/M2/M3/M4)
+
+> ⚠️ Avoid using either system-installed Python or Python from Homebrew, as these versions are often labeled `--externally-managed` resulting in restricted dependency installation.
+
+### ✅ Recommended Setup (using pyenv):
+
+1. **Install pyenv**:
+
+```bash
+brew install pyenv
+```
+
+2. **Install Python**:
+
+```bash
+pyenv install 3.11.9
+pyenv global 3.11.9
+```
+
+3. **Check Python version**:
+
+```bash
+python --version
+```
+
+4. **Create and Activate a Virtual Environment**: Since Apache Airflow requires multiple dependencies, it's a good practice to isolate these dependencies in a virtual environment.
+
+- Create a virtual environment:
+
+```bash
+python -m venv airflow_venv
+```
+
+- Activate the virtual environment:
+
+```bash
+source airflow_venv/bin/activate
+```
+
+5. **Install Apache Airflow**: Apache Airflow is available on PyPI. To install it, you can use the following command in your terminal:
+
+```bash
+pip install apache-airflow
```

Comment:
```suggestion
pip install apache-airflow==3.0.0 --constraint "https://raw.githubusercontent.com/apache/airflow/constraints-3.0.0/constraints-3.11.txt"
```

Note that installing with constraints - at least initially - is recommended for reproducible installation. It might sometimes happen that 3rd-party distributions are relased and their latest versions break airflow. Using constraints makes the installation reproducible with versions of dependencies that were "frozen" at the time of releasing airflow. Note you have to specify both - airflow version and Python version you are using.